mux.go

```diff
@@ -36,7 +36,7 @@ type Mux struct {
 type MuxConfig struct {
 	KeyRing         keychain.SecretKeyRing
 	ActiveNetParams *chaincfg.Params
-	SettledCallback func(lntypes.Hash)
+	SettledCallback func(lntypes.Hash, types.Status)
```

Comment:
Maybe make it `types.InvoiceStatus`. Otherwise confusion will happen with the next status type added.

Reply:
Yeah I do agree on the separation

To be consistent with what we did in the DB, I will create 2 distinct types (not aliasing otherwise it can be confusing to have a `HTLCStatus` with  type `InvoiceStatus`)

It will ease things if one day we want to change/add a status for either invoice or htlc